examples/stable-diffusion/training/README.md

```diff
@@ -61,9 +61,94 @@ python textual_inversion.py \
   --throughput_warmup_steps 3
 ```
 
+The following example shows how to run inference using the fine-tuned model:
+
+```python
+from optimum.habana.diffusers import GaudiStableDiffusionPipeline
+import torch
+
+model_id = "/tmp/textual_inversion_cat"
+pipe = GaudiStableDiffusionPipeline.from_pretrained(
+    model_id,
+    torch_dtype=torch.bfloat16,
+    use_habana=True,
+    use_hpu_graphs=True,
+    gaudi_config="Habana/stable-diffusion",
+)
+
+prompt = "A <cat-toy> backpack"
+image = pipe(prompt, num_inference_steps=50, guidance_scale=7.5).images[0]
+image.save(f"cat-backpack.png")
+```
+
 > Change `--resolution` to 768 if you are using the [stable-diffusion-2](https://huggingface.co/stabilityai/stable-diffusion-2) 768x768 model.
 
-> As described in [the official paper](https://arxiv.org/abs/2208.01618), only one embedding vector is used for the placeholder token, *e.g.* `"<cat-toy>"`. However, one can also add multiple embedding vectors for the placeholder token to increase the number of fine-tuneable parameters. This can help the model to learn more complex details. To use multiple embedding vectors, you can define `--num_vectors` to a number larger than one, *e.g.*: `--num_vectors 5`. The saved textual inversion vectors will then be larger in size compared to the default case.
+> As described in [the official paper](https://arxiv.org/abs/2208.01618), only one embedding vector is used for the placeholder token, *e.g.* `"<cat-toy>"`.
+> However, one can also add multiple embedding vectors for the placeholder token to increase the number of fine-tuneable parameters.
+> This can help the model to learn more complex details. To use multiple embedding vectors, you can define `--num_vectors` to a number larger than one,
+> *e.g.*: `--num_vectors 5`. The saved textual inversion vectors will then be larger in size compared to the default case.
+
+
+## Textual Inversion XL
+
+The `textual_inversion_sdxl.py` script shows how to implement textual inversion fine-tuning on Gaudi for XL diffusion models
+such as `stabilityai/stable-diffusion-xl-base-1.0` or `cagliostrolab/animagine-xl-3.1` for example.
+
+Assuming the afforemenioned cat toy dataset has been obtained, we can launch textual inversion XL training using:
+
+```bash
+python textual_inversion_sdxl.py \
```

Comment:
I am getting this error running this. thoughts?
```bash 
{'dynamic_thresholding_ratio', 'thresholding', 'variance_type', 'clip_sample_range', 'rescale_betas_zero_snr'} was not found in config. Values will be initialized to default values.
08/20/2024 23:45:37 - INFO - __main__ - ***** Running training *****
08/20/2024 23:45:37 - INFO - __main__ -   Num examples = 700
08/20/2024 23:45:37 - INFO - __main__ -   Num Epochs = 3
08/20/2024 23:45:37 - INFO - __main__ -   Instantaneous batch size per device = 1
08/20/2024 23:45:37 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 4
08/20/2024 23:45:37 - INFO - __main__ -   Gradient Accumulation steps = 4
08/20/2024 23:45:37 - INFO - __main__ -   Total optimization steps = 500
Steps:   0%|                                                                                                                                         | 0/500 [00:00<?, ?it/s]/usr/local/lib/python3.10/dist-packages/diffusers/models/unets/unet_2d_blocks.py:1369: FutureWarning: `scale` is deprecated and will be removed in version 1.0.0. The `scale` argument is deprecated and will be ignored. Please remove it, as passing it will raise an error in the future. `scale` should directly be passed while calling the underlying pipeline component i.e., via `cross_attention_kwargs`.
  deprecate("scale", "1.0.0", deprecation_message)
/usr/local/lib/python3.10/dist-packages/diffusers/models/unets/unet_2d_blocks.py:2628: FutureWarning: `scale` is deprecated and will be removed in version 1.0.0. The `scale` argument is deprecated and will be ignored. Please remove it, as passing it will raise an error in the future. `scale` should directly be passed while calling the underlying pipeline component i.e., via `cross_attention_kwargs`.
  deprecate("scale", "1.0.0", deprecation_message)
Steps:   1%|▊                                                                                                        | 4/500 [03:40<4:39:55, 33.86s/it, loss=0.287, lr=0.002]Traceback (most recent call last):
  File "/devops/sgohari/tests/codes/pr-reviews/868/optimum-habana/examples/stable-diffusion/training/textual_inversion_sdxl.py", line 1076, in <module>
    main()
  File "/devops/sgohari/tests/codes/pr-reviews/868/optimum-habana/examples/stable-diffusion/training/textual_inversion_sdxl.py", line 851, in main
    for step, batch in enumerate(train_dataloader):
  File "/usr/local/lib/python3.10/dist-packages/accelerate/data_loader.py", line 464, in __iter__
    next_batch = next(dataloader_iter)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 631, in __next__
    data = self._next_data()
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 675, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/devops/sgohari/tests/codes/pr-reviews/868/optimum-habana/examples/stable-diffusion/training/textual_inversion_sdxl.py", line 519, in __getitem__
    image = Image.open(self.image_paths[i % self.num_images])
  File "/usr/local/lib/python3.10/dist-packages/PIL/Image.py", line 3431, in open
    fp = builtins.open(filename, "rb")
IsADirectoryError: [Errno 21] Is a directory: '/devops/sgohari/tests/codes/pr-reviews/868/optimum-habana/examples/stable-diffusion/training/cat/.cache'
Steps:   1%|▊                                                                                                        | 4/500 [03:40<7:35:26, 55.09s/it, loss=0.287, lr=0.002]
```

Reply:
I had removed this folder and redownloed but got the same error. 
Not sure what happened but removing that .cache folder, worked. thanks.